cli/module_graph.rs

```diff
@@ -667,6 +688,41 @@ pub struct Graph {
   maybe_lockfile: Option<Arc<Mutex<Lockfile>>>,
 }
 
+/// Convert a specifier and a module slot in a result to the module source which
+/// is needed by Deno core for loading the module.
+fn to_module_result(
+  (specifier, module_slot): (&ModuleSpecifier, &ModuleSlot),
+) -> (ModuleSpecifier, Result<ModuleSource, AnyError>) {
+  match module_slot {
+    ModuleSlot::Err(err) => (specifier.clone(), Err(anyhow!(err.to_string()))),
+    ModuleSlot::Module(module) => (
+      specifier.clone(),
+      if let Some(emit) = &module.maybe_emit {
+        match emit {
+          Emit::Cli((code, _)) => Ok(ModuleSource {
+            code: code.clone(),
+            module_url_found: module.specifier.to_string(),
+            module_url_specified: specifier.to_string(),
+          }),
+        }
+      } else {
+        match module.media_type {
+          MediaType::JavaScript | MediaType::Unknown => Ok(ModuleSource {
+            code: module.source.clone(),
+            module_url_found: module.specifier.to_string(),
+            module_url_specified: specifier.to_string(),
+          }),
+          _ => Err(anyhow!("Compiled module not found \"{}\"", specifier)),
```

Comment:
Using `anyhow!` will result in generic `Error` class, shouldn't we use `NotFound` instead here?